src/provider/utils/async.ts

```diff
@@ -78,3 +78,23 @@ export function allowReject<T>(promise: Promise<T>): Promise<T> {
     promise.catch(() => {});
     return promise;
 }
+
+export async function asyncFilter<T>(arr: T[], asyncF: (x: T) => Promise<boolean>): Promise<T[]> {
```

Comment:
think the symbol `asyncF` sounds a tiny bit ambiguous especially in a context where method name is `asyncFilter`. should this be `asyncFunc `?

Reply:
Gone with `callback`, following https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter